browser/download/brave_download_service_factory.h

```diff
@@ -0,0 +1,49 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_BROWSER_DOWNLOAD_BRAVE_DOWNLOAD_SERVICE_FACTORY_H_
+#define BRAVE_BROWSER_DOWNLOAD_BRAVE_DOWNLOAD_SERVICE_FACTORY_H_
+
+#include <memory>
+
+#include "base/macros.h"
+#include "components/keyed_service/core/simple_keyed_service_factory.h"
+
+class SimpleFactoryKey;
+
+namespace base {
+template <typename T>
+struct DefaultSingletonTraits;
+}  // namespace base
+
+namespace download {
+class DownloadService;
+}  // namespace download
+
+// BraveDownloadServiceFactory is the main client class for interaction with the
+// download component.
+class BraveDownloadServiceFactory : public SimpleKeyedServiceFactory {
```

Comment:
why did you have to copy the factory?..

Reply:
It would have required a lot of patching to use the existing Chromium `DownloadServiceFactory`, I think it was due to `DownloadClientMap` and a few other signatures.